testutils/resourcebuilder/config/v1/infrastructure.go

```diff
@@ -382,3 +393,20 @@ func (i InfrastructureBuilder) WithNamespace(namespace string) InfrastructureBui
 	i.namespace = namespace
 	return i
 }
+
+// WithPlatformStatus sets the platformStatus for the infrastructure builder.
+func (i InfrastructureBuilder) WithPlatformStatus(ps configv1.PlatformStatus) InfrastructureBuilder {
+	if i.status == nil {
+		i.status = &configv1.InfrastructureStatus{}
+	}
+
+	i.status.PlatformStatus = &ps
+
+	return i
+}
+
+// WithNamespace sets the status for the infrastructure builder.
```

Comment:
update the doc string here, also do we need a nil check here as well?

Reply:
Ahh thanks for catching this. Fixed.